source/server/server.cc

```diff
@@ -452,11 +455,17 @@ void InstanceImpl::initialize(const Options& options,
   const bool use_tcp_for_dns_lookups = bootstrap_.use_tcp_for_dns_lookups();
   dns_resolver_ = dispatcher_->createDnsResolver({}, use_tcp_for_dns_lookups);
 
+  // the load stats reporter needs a custom stats root for independent merge intervals
+  if (bootstrap_.cluster_manager().has_load_stats_config()) {
+    load_report_stats_store_ = std::make_unique<Stats::ThreadLocalStoreImpl>(*allocator_);
+    load_report_stats_store_->initializeThreading(*dispatcher_, thread_local_);
```

Comment:
Why does the ownership of load report stats store need to live in the server object? Can it be localized to ClusterManager? That might reduce a bit of threading.

Reply:
Yeah, there's no good reason other than `ClusterManager` traditionally didn't seem to be doing things like `initializeThreading`. I'll move it there.